instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware.py

```diff
@@ -156,12 +192,36 @@ def process_request(self, request):
             ),
         )
 
-        attributes = collect_request_attributes(request_meta)
+        if is_asgi_request:
+            attributes = collect_request_attributes(request.scope)
+        else:
+            attributes = collect_request_attributes(request_meta)
 
         if span.is_recording():
-            attributes = extract_attributes_from_object(
-                request, self._traced_request_attrs, attributes
-            )
+            if is_asgi_request:
+                # ASGI requests include extra attributes in request.scope.headers. For this reason,
+                # we need to build an object with the union of `request` and `request.scope.headers`
+                # contents, for the extract_attributes_from_object function to be able to retrieve
+                # attributes from it.
+                attributes = extract_attributes_from_object(
+                    types.SimpleNamespace(
+                        **{
+                            **request.__dict__,
+                            **{
+                                name.decode("latin1"): value.decode("latin1")
```

Comment:
Just curious, why `"latin1"`?

Reply:
Here, I'm using the same logic Django uses to decode the request scope in ASGI mode, which uses `latin1`. This is the relevant upstream code: https://github.com/django/django/blob/7bb0605a8c4adb4247a179276036f138ae859ac9/django/core/handlers/asgi.py#L76-L90